odltools/karaf/dump.py

```diff
@@ -0,0 +1,140 @@
+# Copyright 2018 Red Hat, Inc. and others. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import sys
+import os
+
+line_num = 0
+OPENER = '[{<'
+CLOSER = ']}>'
+INDENT = '  '
+SKIP_FIELDS = 4
+
+
+def dump_karaf_log(args):
+    path = args.path
+    if path == '-':
+        _dump_pretty_print(sys.stdin)
+        return
+
+    if not os.path.isfile(path):
+        sys.stderr.write('File "%s" not found' % path)
```

Comment:
use logging.error and also raise an exception class like InvalidKarafArgument("Karaf log file does not exist")

Reply:
I would also put this in the argument parsing method, as this arg may later be used for other methods

Reply:
Is this not the argument parsing method? Or do you mean adding a custom validator (type) when the parser is built?

Reply:
yeah the 2nd one.  Since you might use the log to do more actions other than your "dump" one, you can just do the validation of the input in the cli.py.  Other than that I think this is fine to approve/merge.